lib/client.js

```diff
@@ -2,11 +2,13 @@
 
 import { Response } from './response';
 
+const $request = Symbol('request');
+
 export class Client {
-  constructor(config={}) {
-    // config.url
-    // config.token
-    // todo: set up properties
+  constructor({ url, token, request } = {}) {
+    assertFunction(request);
+
+    this[$request] = request;
```

Comment:
The challenge with some sort of singleton is that it can only be so for the current instance of `client` rather than universal, so you can't statically import it as a module.  Otherwise you couldn't use two different sdks built on JSClient at the same time without them sharing the config.

The closest thing we have to an object that is accessible *everywhere* is the client object itself, which is passed to models, collections, and likely requests.

Also, in practice, I don't think passing things like this around is going to bite us.  Other than `Client` itself, end users of sdks shouldn't need to manually instantiate anything.

That all said, I'm open to the idea if you can come up with something.